docs/content/operations/tls-support.md

```diff
@@ -0,0 +1,50 @@
+and `druid.tlsPort` properties on each node. Please see `Configuration` section of individual nodes to check the valid and default values for these ports.
+
+# Jetty Server TLS Configuration
+
+Druid uses Jetty as an embedded web server. To get familiar with TLS/SSL in general and related concepts like Certificates etc.
+reading this [Jetty documentation](http://www.eclipse.org/jetty/documentation/9.3.x/configuring-ssl.html) might be helpful.
+To get more in depth knowledge of TLS/SSL support in Java in general, please refer to this [guide](http://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html).
+The documentation [here](http://www.eclipse.org/jetty/documentation/9.3.x/configuring-ssl.html#configuring-sslcontextfactory)
+can help in understanding TLS/SSL configurations listed below. This [document](http://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html) lists all the possible
+values for the below mentioned configs among others provided by Java implementation.
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+|`druid.server.https.keyStorePath`|The file path or URL of the TLS/SSL Key store.|`null`|yes|
+|`druid.server.https.keyStoreType`|The type of the key store.|`null`|yes|
+|`druid.server.https.certAlias`|Alias of TLS/SSL certificate for the connector.|`null`|yes|
+
+All these configs must be set to some valid value. Apart from these properties, `keyStorePassword` and `keyManagerPassword` are required
+to read Certificate from the Key Store. These passwords will be fetched using a `PasswordProvider` implementation.
+For example, `DefaultPasswordProvider` will look for `druid.password.keyStorePassword` and `druid.password.keyManagerPassword` runtime properties
+to get these passwords. `EnvironmentVariablePasswordProvider` will directly look for values of `keyStorePassword` and `keyManagerPassword` environment variables.
+If some custom `PasswordProvider` is being used, please make sure `getPassword(String key)` method can (optionally securely) fetch these passwords.
+
+# Druid's internal communication over TLS
+
+Whenever possible Druid nodes will use HTTPS to talk to each other. To enable this communication Druid's HttpClient need to
+be configured with a proper [SSLContext](http://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLContext.html) that is able
+to validated the Server Certificates, otherwise communication will fail.
+
+Since, there are various ways to configure SSLContext, by default, Druid looks for an instance of SSLContext Guice binding
+while creating the HttpClient. This binding can be achieved writing a [Druid extension](../development/extensions.html)
+which can provide an instance of SSLContext. Druid comes with a sample extension present [here](../development/extensions-core/sample-client-sslcontext.html)
+which should be useful enough for most simple cases, see [this](./including-extensions.html) for how to include extensions.
```

Comment:
since this extension is meant to be used in production for simple cases, I agree with @nishantmonu51 on removing "sample" (or maybe change it to "simple")